File: C++/Manual 05/Question05.cpp
#include <iostream>
using namespace std;

int main()
{
    // Question 5
    double temp = 0;
    int time = 0;
    cout << "Enter The Temperature for the Time Elapsed in Minutes" << endl;
    cout << "To End Program, Enter Temp as -1" << endl
         << endl;
    for (int i = 1; temp != -1; i++)
    {
        time += 15;
        cout << "Enter the Temperature. Time Elapsed: " << time << " Minutes." << endl;
        cout << "Temperature(C): ";
        cin >> temp;
    tempcheck:
        if (temp > 102.5)
        {
            cout << "Temperature has risen above Critical Limit! Let it Cool for 5 Minutes and Enter The Temperature Again!" << endl
                 << endl;
            time += 5;
            cout << "Time Elapsed: " << time << " Minutes. Enter the Temperature." << endl;
            cout << "Temperature: ";
            cin >> temp;
            goto tempcheck;
        }
    }
    return 0;
}

File: Python/Manual 05/Question05.py
import time

# 1 sec represents 5mins


def check_temperature(temp):
    while temp > 102.5:
        print(f"Temperature is {temp}°C, which is above the threshold.")
        print("Turning down the thermostat...")
        time.sleep(1)

        temp -= 1.0
        print(f"Rechecking temperature: {temp}°C")

    print(f"Temperature is now {temp}°C. Technician does nothing.")


while True:
    temp = float(input("Enter the current temperature in Celsius: "))
    if temp <= 102.5:
        print(f"Temperature is {temp}°C. Technician does nothing.")
    else:
        check_temperature(temp)
        print("Waiting for 15 minutes before the next check...")
        time.sleep(3)
